readthedocs/core/unresolver.py

```diff
@@ -1,99 +1,284 @@
-            # We need to add the index.html to find this actual file
-            filename += 'index.html'
-
-        log.debug(
-            'Unresolver parsed.',
-            project_slug=final_project.slug,
-            lang_slug=lang_slug,
-            version_slug=version_slug,
-            filename=filename,
+    def _unresolve_path(self, parent_project, path, check_subprojects=True):
+        """
+        Unresolve `path` with `parent_project` as base.
+
+        If the returned project is `None`, then we weren't able to
+        unresolve the path into a project.
+
+        If the returned version is `None`, then we weren't able to
+        unresolve the path into a valid version of the project.
+
+        :param parent_project: The project that owns the path.
+        :param path: The path to unresolve.
+        :param check_subprojects: If we should check for subprojects,
+         this is used to call this function recursively.
+
+        :returns: A tuple with: project, version, and file name.
+        """
+        # Multiversion project.
+        if not parent_project.single_version:
+            response = self._match_multiversion_project(
+                parent_project=parent_project,
+                path=path,
+            )
+            if response:
+                return response
+
+        # Subprojects.
+        if check_subprojects:
+            response = self._match_subproject(
+                parent_project=parent_project,
+                path=path,
+            )
+            if response:
+                return response
+
+        # Single version project.
+        if parent_project.single_version:
+            response = self._match_single_version_project(
+                parent_project=parent_project,
+                path=path,
+            )
+            if response:
+                return response
+
+        return None, None, None
+
+    @staticmethod
+    def get_domain_from_host(host):
+        """
+        Get the normalized domain from a hostname.
+
+        A hostname can include the port.
+        """
+        return host.lower().split(":")[0]
+
+    # TODO: make this a private method once
+    # proxito uses the unresolve method directly.
+    def unresolve_domain(self, domain):
+        """
+        Unresolve domain by extracting relevant information from it.
+
+        :param str domain: Domain to extract the information from.
+        :returns: A tuple with: the project slug, domain object, and if the domain
+         is from an external version.
+        """
+        public_domain = self.get_domain_from_host(settings.PUBLIC_DOMAIN)
+        external_domain = self.get_domain_from_host(
+            settings.RTD_EXTERNAL_VERSION_DOMAIN
         )
-        return UnresolvedObject(final_project, lang_slug, version_slug, filename, parsed.fragment)
 
+        subdomain, *root_domain = domain.split(".", maxsplit=1)
+        root_domain = root_domain[0] if root_domain else ""
```

Comment:
I think it's not required use `*root_domain` here. Why are you converting it to a list of one element? Since `maxsplit=1` it will always return a list of 2 elements here.

```
In [10]: subdomain, root_domain = 'docs.readthedocs.io'.split('.', maxsplit=1)

In [11]: subdomain
Out[11]: 'docs'

In [12]: root_domain
Out[12]: 'readthedocs.io'
```

Reply:
It returns only one element if the string doesn't have a `.`.